integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/predicate/NotPredicateBaseIT.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.integrationtest.backend.tck.search.predicate;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.hibernate.search.engine.search.predicate.dsl.PredicateFinalStep;
+import org.hibernate.search.engine.search.predicate.dsl.SearchPredicateFactory;
+import org.hibernate.search.integrationtest.backend.tck.testsupport.util.rule.SearchSetupHelper;
+import org.hibernate.search.util.impl.integrationtest.mapper.stub.BulkIndexer;
+import org.hibernate.search.util.impl.integrationtest.mapper.stub.StubMappedIndex;
+import org.hibernate.search.util.impl.test.runner.nested.Nested;
+import org.hibernate.search.util.impl.test.runner.nested.NestedRunner;
+
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(NestedRunner.class)
+public class NotPredicateBaseIT {
+
+	@ClassRule
+	public static SearchSetupHelper setupHelper = new SearchSetupHelper();
+
+	@BeforeClass
+	public static void setup() {
+		setupHelper.start()
+				.withIndexes(
+						ScoreIT.index
+				)
+				.setup();
+
+		final BulkIndexer scoreIndexer = ScoreIT.index.bulkIndexer();
+		ScoreIT.dataSet.contribute( scoreIndexer );
+
+		scoreIndexer.join();
+	}
+
+	@Test
+	public void takariCpSuiteWorkaround() {
+		// Workaround to get Takari-CPSuite to run this test.
+	}
+
+	@Nested
+	public static class ScoreIT extends AbstractPredicateScoreIT {
+		private static final DataSet dataSet = new DataSet();
+
+		private static final StubMappedIndex index = StubMappedIndex.withoutFields().name( "score" );
+
+		public ScoreIT() {
+			super( index, dataSet );
+		}
+
+
+		@Override
+		public void predicateLevelBoost() {
+			// ignore
```

Comment:
good question 😃. with ES this test ultimately generates a query (I've removed routing etc..):
```json
{
  "query": {
    "bool": {
      "must": {
        "bool": {
          "should": [
            {
              "bool": {
                "must_not": {
                  "ids": {
                    "values": [
                      "singleRoutingKey_doc_1",
                      "singleRoutingKey_doc_2"
                    ]
                  }
                }
              }
            },
            {
              "bool": {
                "boost": 7.0,
                "must_not": {
                  "ids": {
                    "values": [
                      "singleRoutingKey_doc_0",
                      "singleRoutingKey_doc_2"
                    ]
                  }
                }
              }
            }
          ]
        }
      }
    }
  }
}
```
but the results have 0 score:
```json
{
  "total": {
    "value": 2,
    "relation": "eq"
  },
  "max_score": 0.0,
  "hits": [
    {
      "_index": "score-000001",
      "_id": "singleRoutingKey_doc_1",
      "_score": 0.0,
      "_routing": "singleRoutingKey",
      "_source": {
        "_entity_type": "scoreType"
      }
    },
    {
      "_index": "score-000001",
      "_id": "singleRoutingKey_doc_0",
      "_score": 0.0,
      "_routing": "singleRoutingKey",
      "_source": {
        "_entity_type": "scoreType"
      }
    }
  ]
}
```
So I'm not sure how we could make use of a combo of `not` and `boost` for this test to pass ...  Same is with Lucene.

Reply:
Sooo... are you telling me that the score is always 0, thus `constantScore` and `boost` are pointless? If that's the case, that changes everything: better remove those features from the `not` predicate altogether.

Reply:
I guess we could add a `must` so that the scoring is enabled. Then the query would become something like this:
```json
{
  "query": {
    "bool": {
      "must": {
        "bool": {
          "should": [
            {
              "bool": {
                "must": {
                  "match_all": {}
                },
                "must_not": {
                  "ids": {
                    "values": [
                      "singleRoutingKey_doc_1",
                      "singleRoutingKey_doc_2"
                    ]
                  }
                }
              }
            },
            {
              "bool": {
                "boost": 7.0,
                "must": {
                  "match_all": {}
                },
                "must_not": {
                  "ids": {
                    "values": [
                      "singleRoutingKey_doc_0",
                      "singleRoutingKey_doc_2"
                    ]
                  }
                }
              }
            }
          ]
        }
      }
    }
  }
}
```
but then the "optimizations" would become a bit more cumbersome 😃. I guess ... if this `not` is intended for those who aren't familiar with `bool` would it be safe to assume that they wouldn't need the boost anyway? (I'll give it a try with the `match_all` in the meantime).

Reply:
> We can add the implicit `match_all` _after_ the optimizations, so that would be simpler.

By that, I meant much after, when we generate JSON.